app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -384,6 +384,36 @@ protected function getRatesDebuggingData($rates) {
         return $rateDebuggingData;
     }
 
+    /**
+     * Removes a bolt order from the system.  The order is expected to be a Bolt order
+     *
+     * @param Mage_Sales_Model_Order $order
+     *
+     * @throws Mage_Core_Exception if the order cannot be canceled
+     */
+    public function removePreAuthOrder($order) {
+        if ($this->isBoltOrder($order)) {
+            if ($order->getStatus() !== 'canceled_bolt') {
+                $order->cancel()->setQuoteId(null)->setStatus('canceled_bolt')->save();
+            }
+            $previousStoreId = Mage::app()->getStore()->getId();
+            Mage::app()->setCurrentStore(Mage_Core_Model_App::ADMIN_STORE_ID);
```

Comment:
why you need this?

Reply:
@daisy1754 
I see that you have a question about something but I am not sure what you have a question about.  I also do not see a change request here, but rather you desire an explanation of something.

When you say:
> why you need this?

What is the `this` that you are referring to? 
_personally, I only seem to need air, food, and water.  :)_

Reply:
why do you need to set store to admin store?

Reply:
@daisy1754 
We must set the store to admin in order to delete the order.  In Magento, the admin area is required to delete this type of entity.  You may review the `Mage_Sales_Model_Order` class and trace the core code back to observe this.  Eventually, you will find this passage:
https://github.com/OpenMage/magento-mirror/blob/magento-1.9/app/code/core/Mage/Sales/Model/Order.php#L2322-L2331

From there, you will see that this code is invoked from an Abstract method:
```
    /**
     * Safeguard func that will check, if we are in admin area
     *
     * @throws Mage_Core_Exception
     */
    protected function _protectFromNonAdmin()
    {
        if (Mage::registry('isSecureArea')) {
            return;
        }
        if (!Mage::app()->getStore()->isAdmin()) {
            Mage::throwException(Mage::helper('core')->__('Cannot complete this operation from non-admin area.'));
        }
    }
```

Also, this will be true for Magento 2:
https://github.com/OpenMage/magento-mirror/blob/magento-1.9/app/code/core/Mage/Sales/Model/Order.php#L2322-L2331